src/server/middleware/util.go

```diff
@@ -70,50 +67,16 @@ type ManifestInfo struct {
 // ManifestExists ...
 func (info *ManifestInfo) ManifestExists(ctx context.Context) (bool, error) {
 	info.manifestExistOnce.Do(func() {
-
-		// ToDo: use the artifact controller method
-		total, repos, err := repository.Mgr.List(ctx, &q.Query{
-			Keywords: map[string]interface{}{
-				"Name": info.Repository,
-			},
-		})
-		if err != nil {
-			info.manifestExistErr = err
-			return
-		}
-		if total == 0 {
-			return
-		}
-
-		total, tags, err := tag.Mgr.List(ctx, &q.Query{
-			Keywords: map[string]interface{}{
-				"Name":         info.Tag,
-				"RepositoryID": repos[0].RepositoryID,
-			},
-		})
+		af, err := artifact.Ctl.GetByReference(ctx, info.Repository, info.Tag, nil)
 		if err != nil {
 			info.manifestExistErr = err
 			return
 		}
-		if total == 0 {
+		if af == nil {
```

Comment:
`af` will never be `nil` return by `artifact.Ctl.GetByReference` when `err` is not `nil`.

Reply:
ditto